webhook/plan_webhook.go

```diff
@@ -0,0 +1,116 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var kubevirtprojectsClusterViewGVR = schema.GroupVersionResource{
+	Group:    "clusterview.open-cluster-management.io",
+	Version:  "v1",
+	Resource: "kubevirtprojects",
+}
+
+func ValidateWebhook(c client.Client, config rest.Config) *webhook.Admission {
+	return &webhook.Admission{
+		Handler: admission.HandlerFunc(func(ctx context.Context, req webhook.AdmissionRequest) webhook.AdmissionResponse {
+			log := ctrl.LoggerFrom(ctx).WithValues("operation", req.Operation, "user", req.UserInfo.Username)
+			if req.Operation == v1.Create || req.Operation == v1.Update {
+				if len(req.Object.Raw) == 0 {
+					return webhook.Denied("Request object is empty")
+				}
+
+				plan, err := rawToPlan(req.Object)
+				if plan == nil || err != nil {
+					log.Error(err, "Failed to parse request object into Plan")
+					return webhook.Denied("Failed to parse request object into Plan")
+				}
+
+				targetNamespace := plan.Spec.TargetNamespace
+				clusterName := plan.Spec.Provider.Destination.Name
+
+				log = log.WithValues("cluster", clusterName, "namespace", targetNamespace)
+
+				config.Impersonate = rest.ImpersonationConfig{
+					UserName: req.UserInfo.Username,
+					Groups:   req.UserInfo.Groups,
+					UID:      req.UserInfo.UID,
+				}
+
+				dynamicClient, err := dynamic.NewForConfig(&config)
+				if err != nil {
+					log.Error(err, "Failed to initialize dynamic client with impersonation")
+					return webhook.Denied("Failed to setup dynamic client")
+				}
+
+				valid, err := validateKubevirtView(dynamicClient, clusterName, targetNamespace)
+				if err != nil {
+					log.Error(err, "Validation failed during kubevirtprojects access check")
+					return webhook.Denied("Failed to validate access to kubevirtprojects")
```

Comment:
user is not going to understand what kubevirtprojects is.  This is related to comments below.